lib/csr.cpp

```diff
@@ -164,7 +166,19 @@ Reg Csr::implicit_read(Csr::Address csr) {//Does not perform any privilege check
         default: rv_trap::invoke_exception(rv_trap::Cause::ILLEGAL_INSTRUCTION_EXCEPTION);
     }
 
-    return 0; // Avoid no return warning
+    assert(false && "We should never get here");
+    std::unreachable();
+}
+
+Csr::interrupt_regs Csr::fast_implicit_read_interrupt_regs() const {
```

Comment:
Yeah I was thinking the same thing. The interrupt checks where the largest offenders in terms of performance, which is why I prioritized them, but you're right it's not very orthogonal in terms of the interface.

I could split this into a bunch of individual getters (I don't actually need any setters), but given how much GCC was struggling to inline the function as-is, it might hurt performance.

If you're okay with it I think perhaps we perhaps leave that improvement for a later date.

Reply:
Yeah, that comment was kind of just more brainstorming. We can definitely leave this for now, and maybe rework later if we do this kind of thing for other spots :)